notebooks/15_flightplanning_tuto.ipynb

```diff
@@ -0,0 +1,1018 @@
+    "    - the current mass\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class State(NamedTuple):\n",
+    "    trajectory: pd.DataFrame\n",
+    "    pos: Tuple[int, int]\n",
+    "\n",
+    "    def __hash__(self):\n",
+    "        return hash(self.pos)\n",
+    "\n",
+    "    def __eq__(self, other):\n",
+    "        return self.pos == other.pos\n",
+    "\n",
+    "    def __ne__(self, other):\n",
+    "        return self.pos != other.pos\n",
+    "\n",
+    "    def __str__(self):\n",
+    "        return f\"[{self.trajectory.iloc[-1]['ts']:.2f} \\\n",
+    "            {self.pos} \\\n",
+    "            {self.trajectory.iloc[-1]['alt']:.2f} \\\n",
+    "            {self.trajectory.iloc[-1]['fuel']:.2f}]\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Action\n",
+    "- The various options the Aircraft can choose to navigate on the graph. In this case, we have three possible actions: UP, DOWN and STRAIGHT"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Action(Enum):\n",
+    "    up = -1\n",
+    "    straight = 0\n",
+    "    down = 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Goal\n",
+    "- The goal is all the points that share the arrival location (latitude, longitude) ARR\n",
+    "\n",
+    "### Reward\n",
+    "\n",
+    "The reward (or cost) associated to any action is the value of the objective function for the given transition (often fuel consumption or time)\n",
+    "\n",
+    "### Starting State\n",
+    "The starting position of the aircraft is the departure airport location (latitude, longitude) DEP. The aircraft is located on the great circle (middle line).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We also specify the type of states, observations, events, transition values, ... \n",
+    "\n",
+    "This is needed so that the various solvers know how to work properly with this domain."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Domain type\n",
+    "\n",
+    "\n",
+    "Then we define the domain type from a base template (`DeterministicPlanningDomain`) with optional refinements (`UnrestrictedActions` and `Renderable`). This corresponds to the following characteristics:\n",
+    "- `DeterministicPlanningDomain`:\n",
+    "    - only one agent\n",
+    "    - deterministic starting state\n",
+    "    - handle only actions\n",
+    "    - actions are sequential\n",
+    "    - deterministic transitions\n",
+    "    - white box transition model\n",
+    "    - goal states are defined\n",
+    "    - positive costs (i.e. negative rewards)\n",
+    "    - fully observable\n",
+    "    - renderable (can be displayed)\n",
```

Comment:
renderable is not in determinisitcPlanningDomain, right ? or if it is it shouldn't